constellations/orion/src/repos/account_repo.go

```diff
@@ -117,32 +118,82 @@ func (acc *accountRepo) SelectAllNegativeBalances() ([]domains.AccountSum, error
 	return results, nil
 }
 
-func (acc *accountRepo) Insert(account domains.Account) error {
+func (acc *accountRepo) InsertWithUser(account domains.Account, user domains.User) error {
+	tx, err := acc.db.Begin()
+	if err != nil {
+		return err
+	}
 	utils.LogWithContext("accountRepo.Insert", logger.Fields{"account": account})
 	statement := "INSERT INTO accounts (" +
 		"created_at, " +
 		"updated_at, " +
 		"primary_email, " +
 		"password" +
 		") VALUES (?, ?, ?, ?)"
-
 	stmt, err := acc.db.Prepare(statement)
 	if err != nil {
+		tx.Rollback()
 		return appErrors.WrapDbPrepare(err, statement)
 	}
 	defer stmt.Close()
 
 	now := time.Now().UTC()
-	execResult, err := stmt.Exec(
+	_, err = stmt.Exec(
 		now,
 		now,
 		account.PrimaryEmail,
 		account.Password,
 	)
 	if err != nil {
-		return appErrors.WrapDbExec(err, statement, account)
+		tx.Rollback()
+		return err
+	}
+
+	statement2 := "INSERT INTO users (" +
+		"created_at, " +
+		"updated_at, " +
+		"first_name, " +
+		"last_name," +
+		"middle_name, " +
+		"email," +
+		"phone, " +
+		"is_guardian," +
+		"account_id," +
+		"notes," +
+		"school," +
+		"graduation_year" +
+		") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+	stmt2, err := acc.db.Prepare(statement2)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	defer stmt2.Close()
+
+	_, err = stmt2.Exec(
+		now,
+		now,
+		user.FirstName,
+		user.LastName,
+		user.MiddleName,
+		user.Email,
+		user.Phone,
+		user.IsGuardian,
+		user.AccountId,
+		user.Notes,
+		user.School,
+		user.GraduationYear,
+	)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Fatal(err)
 	}
-	return appErrors.ValidateDbResult(execResult, 1, "account was not inserted")
+	return appErrors.WrapDbExec(err, statement, account)
```

Comment:
and finally out of the million things that could go wrong, if we get to this point and still no errors, we can just return nil.

Lol sorry about all these if clauses. It's a lot here because we're doing two db operations inside a transaction so there potentially could be a lot wrong here. We are only doing things locally on our computer, so errors are much less likely to happen, but when we launch this live, it's not as simple. The orion webserver and the database could potentially be on their own individual machines crunching data somewhere on Amazon Web Services in one of their data warehouses.

One of the most common things that could happen (why we need all these error checks) is in the middle of our DB operation, orion could lose connectivity to DbMySQL - many reasons, internet sucks, power goes out for a moment, orion server crashes from bad code, or even god forbid, the New York AWS data warehouse gets flooded and all transactions need to route to a different backup database in Toronto or LA.

The operation here probably takes like around a second, but in that time (in computer time), many things could go wrong and we need a way to make sure all cases are handled. Otherwise, you'll end up persisting data when you shouldn't. It is a Go problem to see so many if / err clauses - I'm hoping our new style with appErrors doesn't make it too bad / verbose.